eclair-core/src/main/scala/fr/acinq/eclair/api/Service.scala

```diff
@@ -102,7 +102,12 @@ trait Service extends Logging {
                 case JsonRPCBody(_, _, "send", JString(paymentRequest) :: Nil) =>
                   for {
                     req <- Future(PaymentRequest.read(paymentRequest))
-                    res <- (paymentInitiator ? SendPayment(req.amount.getOrElse(throw new RuntimeException("request without amounts are not supported")).amount, req.paymentHash, req.nodeId)).mapTo[PaymentResult]
+                    res <- (paymentInitiator ? SendPayment(req.amount.getOrElse(throw new RuntimeException("for requests without amount use send (amountMsat, paymentRequet)")).amount, req.paymentHash, req.nodeId)).mapTo[PaymentResult]
```

Comment:
It made explicit the case where amount is not provided, vs the case where we override an existing amount. But that may be overkill, I'll let you choose the version you prefer!